packages/core/node/core-node-server-internal/src/node_config.ts

```diff
@@ -6,33 +6,61 @@
  * Side Public License, v 1.
  */
 
-import { schema } from '@kbn/config-schema';
+import { schema, TypeOf } from '@kbn/config-schema';
 import type { ServiceConfigDescriptor } from '@kbn/core-base-server-internal';
 
 /** @internal */
 export const NODE_CONFIG_PATH = 'node' as const;
+/**
+ * Wildchar is a special config option that implies all {@link NODE_DEFAULT_ROLES} roles.
+ * @internal
+ */
+export const NODE_WILDCARD_CHAR = '*' as const;
+/** @internal */
+export const NODE_BACKGROUND_TASKS_ROLE = 'background_tasks' as const;
+/** @internal */
+export const NODE_UI_ROLE = 'ui' as const;
 /** @internal */
-export const NODE_WILDCARD_CHAR = '*';
+export const NODE_MIGRATOR_ROLE = 'migrator' as const;
 /** @internal */
-export const NODE_ACCEPTED_ROLES = ['background_tasks', 'ui'];
+export const NODE_DEFAULT_ROLES = [NODE_BACKGROUND_TASKS_ROLE, NODE_UI_ROLE] as const;
+/** @internal */
+export const NODE_ALL_ROLES = [
+  NODE_UI_ROLE,
+  NODE_MIGRATOR_ROLE,
+  NODE_BACKGROUND_TASKS_ROLE,
+] as const;
+
+/** @internal */
+export const rolesConfig = schema.oneOf(
+  [
+    schema.arrayOf(
+      schema.oneOf([schema.literal(NODE_BACKGROUND_TASKS_ROLE), schema.literal(NODE_UI_ROLE)])
+    ),
+    schema.arrayOf(schema.literal(NODE_WILDCARD_CHAR), {
+      minSize: 1,
+      maxSize: 1,
+    }),
+    schema.arrayOf(schema.literal(NODE_MIGRATOR_ROLE), { minSize: 1, maxSize: 1 }),
+  ],
+  {
+    defaultValue: [NODE_WILDCARD_CHAR],
+  }
+);
```

Comment:
Using this currently generates a kind of ugly/unhelpful validation error:

```
- [config validation of [node].roles.0.0]: types that failed validation:
 - [config validation of [node].roles.0.0]: expected value to equal [background_tasks]
 - [config validation of [node].roles.0.1]: expected value to equal [ui]
- [config validation of [node].roles.1.0]: expected value to equal [*]
- [config validation of [node].roles.2.1]: expected value to equal [migrator]
```

This is a pre-existing issue, but made slightly worse by adding this piece of configuration.

Reply:
Yeah, ugly config schema errors are a sad reality. But better to have the appropriate config in place.

Reply:
Yeah I like that idea @pgayvallet! This will not cover all cases, like if you pass in a value that is not `*`, `background_tasks` etc. you will still get the old version of the error. But I think this way can be a bit more self-documenting and provide clearer guidance.